File: 0658 find-k-closest-elements/solution.cs
public class Solution {
    public IList<int> FindClosestElements(int[] arr, int k, int x) {
        int l=0,r=arr.Length-1;
        while(r-l+1>k) {
            if(Math.Abs(arr[l]-x) > Math.Abs(arr[r]-x)) l++;
            else r--;
        }
        IList<int> ans = new List<int>();
        for(int i=l;i<=r;i++) ans.Add(arr[i]);
        return ans;
    }
}

File: 0658 find-k-closest-elements/solution.java
class Solution {
    public List<Integer> findClosestElements(int[] nums, int k, int x) {
        int l=0,r=nums.length-1;
        while(r-l+1>k) {
            if(Math.abs(nums[l]-x)>Math.abs(nums[r]-x)) l++;
            else r--;
        }
        List<Integer> ans = new ArrayList<>();
        for(int i=l;i<=r;i++) ans.add(nums[i]);
        return ans;
    }
}
